src/InstrumentationEngine/ProfilerManagerForInstrumentationMethod.cpp

```diff
@@ -29,7 +29,13 @@ CProfilerManagerForInstrumentationMethod::CProfilerManagerForInstrumentationMeth
         // Get InstrumentationMethod-specific LogLevel Environment Variable
         tstring wszInstrumentationMethodLogLevelEnvVar = _T("MicrosoftInstrumentationEngine_LogLevel_") + m_wszInstrumentationMethodGuid;
         WCHAR wszEnvVar[MAX_PATH];
-        if (GetEnvironmentVariable(wszInstrumentationMethodLogLevelEnvVar.c_str(), wszEnvVar, MAX_PATH) > 0)
+        LoggingFlags flags;
+
+        if (SUCCEEDED(pProfilerManager->GetInstrumentationMethodLogLevel(instrumentationMethodGuid, &flags)))
+        {
+            m_flags = flags;
+        }
+        else if (GetEnvironmentVariable(wszInstrumentationMethodLogLevelEnvVar.c_str(), wszEnvVar, MAX_PATH) > 0)
         {
             m_flags = CLoggerService::ExtractLoggingFlags(wszEnvVar);
```

Comment:
Is Error our ultimately default, even if nothing is specified?

Reply:
I believe that None is the default, which I think is the appropriate default value.

---
In reply to: [399476438](https://github.com/microsoft/CLRInstrumentationEngine/pull/231#discussion_r399476438) [](ancestors = 399476438)